packages/create-cloudflare/templates/qwik/pages/c3.ts

```
@@ -0,0 +1,149 @@
+import { endSection } from "@cloudflare/cli";
+import { brandColor } from "@cloudflare/cli/colors";
+import { spinner } from "@cloudflare/cli/interactive";
+import { runFrameworkGenerator } from "frameworks/index";
+import { loadTemplateSnippets, transformFile } from "helpers/codemod";
+import { quoteShellArgs, runCommand } from "helpers/command";
+import { usesTypescript } from "helpers/files";
+import { detectPackageManager } from "helpers/packageManagers";
+import * as recast from "recast";
+import type { TemplateConfig } from "../../../src/templates";
+import type { C3Context } from "types";
+
+const { npm, npx, name } = detectPackageManager();
+
+const generate = async (ctx: C3Context) => {
+	await runFrameworkGenerator(ctx, ["playground", ctx.project.name]);
+};
+
+const configure = async (ctx: C3Context) => {
+	// Add the pages integration
+	// For some reason `pnpx qwik add` fails for qwik so we use `pnpm qwik add` instead.
+	const cmd = [name === "pnpm" ? npm : npx, "qwik", "add", "cloudflare-pages"];
+	endSection(`Running ${quoteShellArgs(cmd)}`);
+	await runCommand(cmd);
+
+	addBindingsProxy(ctx);
+	populateCloudflareEnv();
+};
+
+const addBindingsProxy = (ctx: C3Context) => {
+	// Qwik only has a typescript template atm.
+	// This check is an extra precaution
+	if (!usesTypescript(ctx)) {
+		return;
```

is it OK if this fails silently?

---

i'm not sure how you'd actually reach that state so i think its ok

---

Classically I would put an assertion here:

```
	assert(usesTypescript(ctx), "expected the project to be using Typescript");
```

(from `node:assert`)